src/transformers/models/pegasus_x/modeling_pegasus_x.py

```diff
@@ -0,0 +1,1659 @@
+# coding=utf-8
+# Copyright 2021, Google and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch PEGASUS model."""
+
+import dataclasses
+import math
+import random
+from typing import Optional, Tuple, Union
+
+import numpy as np
+import torch
+import torch.utils.checkpoint
+from torch import nn
+from torch.nn import CrossEntropyLoss
+
+from ...activations import ACT2FN
+from ...modeling_outputs import (
+    BaseModelOutput,
+    BaseModelOutputWithPastAndCrossAttentions,
+    Seq2SeqLMOutput,
+    Seq2SeqModelOutput,
+)
+from ...modeling_utils import PreTrainedModel
+from ...utils import (
+    add_end_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    logging,
+    replace_return_docstrings,
+)
+from .configuration_pegasus_x import PegasusXConfig
+
+
+logger = logging.get_logger(__name__)
+
+_CHECKPOINT_FOR_DOC = "zphang/pegasus-x-base"
+_CONFIG_FOR_DOC = "PegasusXConfig"
+_TOKENIZER_FOR_DOC = "PegasusTokenizer"
+
+
+PEGASUS_X_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "zphang/pegasus-x-base",
+    "zphang/pegasus-x-large",
+    # See all PEGASUS models at https://huggingface.co/models?filter=pegasus-x
+]
+
+
+@dataclasses.dataclass
+class DimensionInfo:
+    """Wrapper for dimension info."""
+
+    B: int  # batch size
```

Comment:
Not really a fan of upper-cased single letter variables. I do believe it's worth writing out exactly those values every time - it makes the code so much more readable. E.g. now every time I see `dim.G` in the code I have to come here to know what it is. 

If possible IMO, it would be really nice if you would always write out the full name (note that 1000 people will read this code file and potentially extend, adapt it where as we have to put in the original effort only once ;-)) 

Happy to do this for you @zphang if you're too busy